spec/settings_spec.cr

```diff
@@ -58,4 +81,29 @@ describe PlaceOS::Driver::Settings do
     settings.raw?(:hash, :hello).should eq("world")
     settings.raw?(:hash, :no_exist).should eq(nil)
   end
+
+  it "should generate JSON schema from objects" do
+    PlaceOS.introspect(Array(String)).should eq({type: "array", items: {type: "string"}})
+    PlaceOS.introspect(SuperArray).should eq({type: "array"})
+    PlaceOS.introspect(AliasTest::Array).should eq({type: "array", items: {type: "string"}})
+    PlaceOS.introspect(Array(Int32)).should eq({type: "array", items: {type: "integer"}})
+    PlaceOS.introspect(Array(Float32)).should eq({type: "array", items: {type: "number"}})
+    PlaceOS.introspect(Array(Bool)).should eq({type: "array", items: {type: "boolean"}})
+    PlaceOS.introspect(Array(JSON::Any)).should eq({type: "array"})
+    PlaceOS.introspect(NamedTuple(steve: String)).should eq({type: "object", properties: {steve: {type: "string"}}, required: ["steve"]})
+    PlaceOS.introspect(TestEnum).should eq({type: "string", enum: ["Bob", "Jane"]})
+    PlaceOS.introspect(Tuple(String, Bool)).should eq({type: "array", items: [{type: "string"}, {type: "boolean"}]})
+    PlaceOS.introspect(SomeType).should eq({anyOf: [
+      {type: "string"},
+      {type: "array", items: [{type: "string"}, {type: "boolean"}]},
+    ]})
+    PlaceOS.introspect(Hash(String, Int32)).should eq({type: "object", additionalProperties: {type: "integer"}})
+    PlaceOS.introspect(SuperHash).should eq({type: "object"})
```

Comment:
would be good to be able to serialise this into json schema like with `Hash(String, Int32)` but this might be troublesome